src/core/obj.js

```diff
@@ -32,578 +32,600 @@ function fetchDestination(dest) {
+    }
+    return shadow(this, 'numPages', obj);
+  }
+
+  get destinations() {
+    const obj = this._readDests(), dests = Object.create(null);
+    if (obj instanceof NameTree) {
+      const names = obj.getAll();
+      for (let name in names) {
+        dests[name] = fetchDestination(names[name]);
+      }
+    } else if (obj instanceof Dict) {
+      obj.forEach(function(key, value) {
+        if (value) {
+          dests[key] = fetchDestination(value);
         }
-      } else if (obj instanceof Dict) {
-        obj.forEach(function(key, value) {
-          if (value) {
-            dests[key] = fetchDestination(value);
-          }
-        });
-      }
-      return shadow(this, 'destinations', dests);
-    },
-    getDestination(destinationId) {
-      const obj = this._readDests();
-      if (obj instanceof NameTree || obj instanceof Dict) {
-        return fetchDestination(obj.get(destinationId) || null);
+      });
+    }
+    return shadow(this, 'destinations', dests);
+  }
+
+  getDestination(destinationId) {
+    const obj = this._readDests();
+    if (obj instanceof NameTree || obj instanceof Dict) {
+      return fetchDestination(obj.get(destinationId) || null);
+    }
+    return null;
+  }
+
+  /**
+   * @private
+   */
+  _readDests() {
+    const obj = this.catDict.get('Names');
+    if (obj && obj.has('Dests')) {
+      return new NameTree(obj.getRaw('Dests'), this.xref);
+    } else if (this.catDict.has('Dests')) { // Simple destination dictionary.
+      return this.catDict.get('Dests');
+    }
+  }
+
+  get pageLabels() {
+    let obj = null;
+    try {
+      obj = this._readPageLabels();
+    } catch (ex) {
+      if (ex instanceof MissingDataException) {
+        throw ex;
       }
+      warn('Unable to read page labels.');
+    }
+    return shadow(this, 'pageLabels', obj);
+  }
+
+  /**
+   * @private
+   */
+  _readPageLabels() {
+    const obj = this.catDict.getRaw('PageLabels');
+    if (!obj) {
       return null;
-    },
-    _readDests() {
-      const obj = this.catDict.get('Names');
-      if (obj && obj.has('Dests')) {
-        return new NameTree(obj.getRaw('Dests'), this.xref);
-      } else if (this.catDict.has('Dests')) { // Simple destination dictionary.
-        return this.catDict.get('Dests');
-      }
-    },
+    }
 
-    get pageLabels() {
-      var obj = null;
-      try {
-        obj = this.readPageLabels();
-      } catch (ex) {
-        if (ex instanceof MissingDataException) {
-          throw ex;
-        }
-        warn('Unable to read page labels.');
-      }
-      return shadow(this, 'pageLabels', obj);
-    },
-    readPageLabels: function Catalog_readPageLabels() {
-      var obj = this.catDict.getRaw('PageLabels');
-      if (!obj) {
-        return null;
-      }
-      var pageLabels = new Array(this.numPages);
-      var style = null;
-      var prefix = '';
+    const pageLabels = new Array(this.numPages);
+    let style = null;
+    let prefix = '';
```

Comment:
Nit: These can probably just go on *one* line instead, i.e. `let style = null, prefix = '';` (since that's how e.g. currentLabel/currentIndex below is already formatted in the old code).